tedana/selection/select_comps.py

```diff
@@ -71,828 +46,298 @@ def selcomps(seldict, comptable, mmix, mask, ref_img, manacc, n_echos, t2s, s0,
-          the acceptance threshold for Kappa values is doubled):
-         A. The variance explained is greater than half the KRcut highest
-             variance component
-        B. Kappa is less than twice Kcut
-    """
-    temp = all_comps[utils.andb([comptable['variance explained'] > 0.5 *
-                                 sorted(comptable['variance explained'])[::-1][int(KRcut)],
-                                 comptable['kappa'] < 2*Kcut]) == 2]
-    KRguess = np.setdiff1d(np.setdiff1d(all_comps[KRelbow == 2], rej),
-                           np.union1d(all_comps[tt_table[:, 0] < tt_lim],
-                           np.union1d(np.union1d(all_comps[spz > 1],
-                                                 all_comps[Vz > 2]),
-                                      temp)))
-    guessmask = np.zeros(len(all_comps))
-    guessmask[KRguess] = 1
-    """
-    Throw lower-risk bad components out based on 3 criteria all being true:
-      1. tt_table (a spatial noisiness metric) <0
-      2. A components variance explains is greater than the median variance
-         explained
-      3. The component index is greater than the KRcut index. Since the
-          components are sorted by kappa, this is another kappa thresholding)
+    Make table of noise gain
     """
-    rejB = acc_comps[utils.andb([tt_table[acc_comps, 0] < 0,
-                                 (comptable['variance explained'][acc_comps] >
-                                  np.median(comptable['variance explained'])),
-                                 acc_comps > KRcut]) == 3]
-    comptable.loc[rejB, 'classification'] = 'rejected'
-    comptable.loc[rejB, 'rationale'] += ('tt_table < 0, component variance '
-                                         'explained > median variance '
-                                         'explained, and component index > '
-                                         'KRcut index;')
-    rej = np.union1d(rej, rejB)
-    # adjust acc_comps again to only contain the remaining non-rejected components
-    acc_comps = np.setdiff1d(acc_comps, rej)
+    comptable['countnoise'] = 0
+    comptable['tt0'] = 0  # needs better name
+    comptable['tt1'] = 0  # needs better name
+    counts_FR2_Z = np.zeros([n_comps, 2])  # unused
+    for i_comp in all_comps:
+        comp_noise_sel = ((np.abs(Z_maps[:, i_comp]) > 1.95) &
+                          (Z_clmaps[:, i_comp] == 0))
+        comptable.loc[i_comp, 'countnoise'] = np.array(
+            comp_noise_sel, dtype=np.int).sum()
+        noise_FR2_Z = np.log10(np.unique(F_R2_maps[comp_noise_sel, i_comp]))
+        signal_FR2_Z = np.log10(np.unique(
+            F_R2_maps[Z_clmaps[:, i_comp] == 1, i_comp]))
+        counts_FR2_Z[i_comp, :] = [len(signal_FR2_Z), len(noise_FR2_Z)]
+        (comptable.loc[i_comp, 'tt0'],
+         comptable.loc[i_comp, 'tt1']) = stats.ttest_ind(
+             signal_FR2_Z, noise_FR2_Z, equal_var=False)
```

Comment:
Given that tt0 and tt1 are t- and p-values, respectively, from a t-test comparing z-transformed R2 F-values from what looks like signal and noise voxels... I think we can maybe called them something like "signal-noise_t" and "signal-noise_p". WDYT?